Cesium3DTilesSelection/src/TilesetContentManager.cpp

```diff
@@ -1162,21 +1169,65 @@ void TilesetContentManager::createLatentChildrenIfNecessary(
   }
 }
 
-bool TilesetContentManager::unloadTileContent(Tile& tile) {
+void TilesetContentManager::finalizeEmptyTile(Tile& tile) {
+  CESIUM_ASSERT(tile.getContent().isEmptyContent());
+  notifyTileUnloading(&tile);
+  tile.getContent().setContentKind(TileUnknownContent{});
+  tile.setState(TileLoadState::Unloaded);
+}
+
+UnloadTileContentResult TilesetContentManager::unloadTileContent(Tile& tile) {
   TileLoadState state = tile.getState();
   if (state == TileLoadState::Unloaded) {
-    return true;
+    return UnloadTileContentResult::Remove;
   }
 
   if (state == TileLoadState::ContentLoading) {
-    return false;
+    return UnloadTileContentResult::Keep;
   }
 
   TileContent& content = tile.getContent();
 
-  // don't unload external or empty tile
-  if (content.isExternalContent() || content.isEmptyContent()) {
-    return false;
+  /**
+   * External tilesets load like this :
+   * 1. Tile in parent tileset points to a new tileset to load. This is the tile
+   *    with TileExternalContent.
+   * 2. TilesetJsonLoader loads new tileset.
+   * 3. The root tile of the new tileset is added as a child of the
+   *    TileExternalContent parent. This tile is TileEmptyContent, but has the
+   *    rest of the tileset as its children.
```

Comment:
> This tile is TileEmptyContent, but has the rest of the tileset as its children.

I don't think this is always true. It depends on whether the root tile of the external tileset.json has content or not. It may be the case that Google Photorealistic 3D Tiles never has content in the root tile of an external tileset, though.